File: c++/e102.cpp
#include <iostream>
using namespace std;

int main()
{
	int num,n,k;
	cin >> num;
	while(num--){
		cin >> n >> k;
		long long ans=1;
		for (int i=n; i>k ;i--) ans*=i;
		for (int i=1;i<=n-k;i++) ans/=i;
		cout << ans << endl;
	}
	return 0;
}

File: python/e102.py
import sys

for num in sys.stdin:
    num=int(num)
    for i in range(num):
        n,k=list(map(int,input().replace('\n','').split(' ')))
        ans=1
        for j in range(n,k,-1):
            ans*=j
        for j in range(1,n-k+1):
            ans/=j
        print (int(ans))
        